hasktorch/src/Torch/Typed/NN/Recurrent/LSTM.hs

```diff
@@ -579,10 +578,8 @@ lstmWithDropout, lstmWithoutDropout
      , hxShape ~ '[numLayers * NumberOfDirections directionality, batchSize, hiddenSize]
      , Parameterized (LSTM inputSize hiddenSize numLayers directionality dtype device) parameters
      , tensorParameters ~ LSTMR inputSize hiddenSize numLayers directionality dtype device
-     , HFoldrM IO TensorListFold [D.ATenTensor] tensorParameters
-     , Apply TensorListUnfold [D.ATenTensor] (HUnfoldMRes IO [D.ATenTensor] tensorParameters)
-     , HUnfoldM IO TensorListUnfold (HUnfoldMRes IO [D.ATenTensor] tensorParameters) tensorParameters
-     , HMap ToDependent parameters tensorParameters
+     , ATen.Castable (HList tensorParameters) [D.ATenTensor]
+     , HMap' ToDependent parameters tensorParameters
```

Comment:
The reduction in constraints here is nice.

Reply:
yes, this was possible before this PR as well, I just didn't see it ;)